src/orion/benchmark/task/task_wrapper.py

```diff
@@ -0,0 +1,164 @@
+from abc import ABC
+from dataclasses import asdict, is_dataclass
+from typing import (
+    Any,
+    Dict,
+    Generic,
+    List,
+    TypeVar,
+    Union,
+)
+import copy
+
+import numpy as np
+from orion.algo.space import Dimension, Space
+from orion.benchmark.task.base import BaseTask, BenchmarkTask
+from logging import getLogger as get_logger
+
+from orion.core.utils.points import flatten_dims, regroup_dims
+from orion.core.utils.format_trials import dict_to_trial
+
+from orion.core.io.space_builder import SpaceBuilder
+from orion.core.utils.format_trials import trial_to_tuple, tuple_to_trial
+from orion.core.worker.trial import Trial
+
+logger = get_logger(__name__)
+
+
+TaskType = TypeVar("TaskType", bound=BaseTask)
+InputType = TypeVar("InputType")
+
+
+class TaskWrapper(BaseTask, Generic[TaskType], ABC):
+    """ ABC for a Wrapper around a Task.
+
+    Wrappers could be used to modify the search space, trials, or the results of the task.
+    """
+
+    def __init__(self, task: TaskType, max_trials: int = None):
+        # TODO: Should Wrappers pass the `task` as an argument to the super() constructor?
+        # When/How are these objects deserialized?
+        if isinstance(task, dict):
+            if len(task) != 1:
+                raise ValueError(
+                    f"Expected task configuration dict to have only one key (the task name), but "
+                    f"got {task} instead."
+                )
+            task_name, task_config = task.popitem()
+            task = BenchmarkTask(of_type=task_name, **task_config)
+        super().__init__(max_trials=max_trials or task.max_trials)
+        self.task: TaskType = task
+
+    def call(self, x: Any) -> List[Dict]:
+        return self.task.call(x)
+
+    def get_search_space(self) -> Dict[str, str]:
+        return self.task.get_search_space()
+
+    @property
+    def unwrapped(self) -> Union[TaskType, BaseTask]:
+        """ Returns the 'unwrapped' task. """
+        return self.task.unwrapped
+
+    @property
+    def configuration(self) -> Dict[str, Any]:
+        """Return the configuration of the task wrapper (including the wrapped task's configuration).
+        """
+        return {
+            type(self).__qualname__: {
+                "task": self.task.configuration,
+                "max_trials": self.max_trials,
+            }
+        }
+
+
+HParamsType = TypeVar("HParamsType")
+
+
+class FixTaskDimensionsWrapper(TaskWrapper[TaskType]):
+    """ Wrapper around a Task that fixes the values of some of its input dimensions.
+    """
+
+    def __init__(self, task: TaskType, fixed_dims: Dict[str, Any], max_trials: int = None):
+        """ Wraps the provided task, fixing the dimensions in `fixed_dims`. """
```

Comment:
No docstrings for `__init__`

Reply:
Huh? Why though? You use the class docstring as the documentation for the constructor?

Reply:
Because the arguments are documented at the level of the class docstring and the docstring of the __init__ method would not bring more information.
We are using numpy's standard: https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes